src/Blazored.Video/VideoQueue.cs

```diff
@@ -0,0 +1,255 @@
+		/// </summary>
+		[Parameter]
+		public EventCallback OnNextPlayed { get; set; }
+
+		/// <summary>
+		///		Will be invoked when there is no other item to play.
+		/// </summary>
+		[Parameter]
+		public EventCallback OnPlaylistEnded { get; set; }
+
+		[CascadingParameter]
+		public BlazoredVideo BlazoredVideo { get; set; }
+
+		public IList<VideoItem> VideoItems { get; }
+		public VideoItem CurrentItem { get; private set; }
+
+		/// <summary>
+		///		Skips the current item and plays the next in queue.
+		/// </summary>
+		/// <returns></returns>
+		public ValueTask PlayNext()
+		{
+			return TryPlayItem(VideoItems
+				.SkipWhile(e => e.Id != CurrentItem.Id)
+				.Skip(1)
+				.FirstOrDefault());
+		}
+
+		/// <summary>
+		///		Skips the current item and plays the previous in queue.
+		/// </summary>
+		/// <returns></returns>
+		public ValueTask PlayPrevious()
+		{
+			return TryPlayItem(VideoItems
+				.Reverse()
+				.SkipWhile(e => e.Id != CurrentItem.Id)
+				.Skip(1)
+				.FirstOrDefault());
+		}
+
+		/// <summary>
+		///		Starts the playback of the given <see cref="VideoItem"/>.
+		/// </summary>
+		/// <param name="value"></param>
+		public async Task Play(VideoItem value)
+		{
+			await BlazoredVideo.PausePlayback();
+			_playAfterRender = true;
+			CurrentItem = value;
+			await OnNextPlayed.InvokeAsync();
+			StateHasChanged();
+		}
+
+		/// <summary>
+		///		Adds a new item into the queue at its end and updates the <see cref="CurrentItem"/> if necessary.
+		/// </summary>
+		/// <param name="videoItem"></param>
+		public void AddVideoItem(VideoItem videoItem)
+		{
+			if (VideoItems.Any(f => f.Id == videoItem.Id))
+			{
+				return;
+			}
+			VideoItems.Add(videoItem);
+			if (VideoItems.Count == 1)
+			{
+				CurrentItem = videoItem;
+			}
+			StateHasChanged();
+		}
+
+		private async ValueTask TryPlayItem(VideoItem next)
+		{
+			await BlazoredVideo.PausePlayback();
+			if (Repeat is RepeatValues.Loop && next is null)
+			{
+				_playAfterRender = true;
+				CurrentItem = VideoItems.FirstOrDefault();
+				StateHasChanged();
+				await OnNextPlayed.InvokeAsync();
+			}
+			else if (next is not null)
+			{
+				_playAfterRender = true;
+				CurrentItem = next;
+				StateHasChanged();
+				await OnNextPlayed.InvokeAsync();
+			}
+			else
+			{
+				await OnPlaylistEnded.InvokeAsync();
+				await BlazoredVideo.PausePlayback();
+			}
+		}
+
+		protected override void OnInitialized()
+		{
+			if (QueueData is { Length: > 0 })
+			{
+				foreach (var queueItem in QueueData)
+				{
+					var item = new VideoItem();
+#pragma warning disable BL0005
+					item.Source = queueItem;
+#pragma warning restore BL0005
+					VideoItems.Add(item);
+				}
+			}
+
+			if (BlazoredVideo.EndedEvent.HasDelegate)
+			{
+				throw new InvalidOperationException("Cannot subscribe to 'BlazoredVideo.EndedEvent' as it is already in use. Please unsubscribe from the event.");
```

Comment:
Maybe "When you are using VideoQueue, you cannot subscribe to `BlazoredVideo.EndedEvent` - you should instead subscribe to `VideoQueue.OnPlaylistEnded`"
... or something similar - to give the dev guidance on the correct solution.